### ISSUE TITLE
Wondering if I should use ng2-redux?

### ISSUE THREAD

**Issue by born2net** [InfomediaLtd/angular2-redux#1]
Just wondering why you didn't use the: https://github.com/wbuchwalter/ng2-redux
is there a benefit to your redux wrapper?

also see here at bottom:
https://github.com/zalmoxisus/redux-devtools-extension/issues/43#issuecomment-177165443

also I think you will be intersted in this:
https://github.com/zalmoxisus/redux-devtools-extension/issues/44

basically they just released a new version of the DevTool with time travel.. sweet...
but with ng2 since it does not refresh, there is a bug, so they asked from Redux team that we apply a fix to your demo app with zone.run() or ApplicationRef.tick() to force update on UI changes AFTER store has updated when time traveling...
regards

Sean

**Comment by rubyboy** [comment]
I actually looked at ng2-redux before creating angular2-redux, but found it a bit more complex that what I wanted. I'm not too keen on the connector approach, especially when it's not clear from the component code what events and properties are connected. It was easier for me to use plain Redux and just wrap it for dependency injection and typings.
Notice that the [AppStore](https://github.com/InfomediaLtd/angular2-redux/blob/master/src/app-store.ts) wrapper in here is very slim. Just a typed wrapper to the Redux app store, with the addition of passing the state to the listeners (Which I thought was important to reduce boilerplate code).

Also, with the addition of the [Actions](https://github.com/InfomediaLtd/angular2-redux/blob/master/src/actions.ts) superclass you could reduce the code even further, by using a "connector-like" approach, but still having explicitly connecting the components events to the dispatch method, like in the [angualr2-redux-example](https://github.com/InfomediaLtd/angular2-redux-example) sample code. For example, the admin component linking the setCurrentUser and setFilmFilter events. The other benefit here is that you don't need to keep the AppStore and actions as properties.
![image](https://cloud.githubusercontent.com/assets/1908166/12700795/cae078e2-c7f9-11e5-86fb-6632ee7c73e0.png)

All that said, if you think ng2-redux is better suited for you, go for it :)

BTW, I've added some comments to the issues you linked. There is no required fix for Redux nor dev tools. It's just the way we bootstrap the app store (and dev tools enhancer). Needs to be inside the Angular zone and now it all works perfectly! I totally dig the dev tools as well :)

**Comment by born2net** [comment]
as always thanks for the effort, and yes, I completely agree, I like your wrapper a lot better as it is simpler and does less magic. When I come back to my code in 6 months using your wrapper, it will be a lot easier to reason on what's happening. The only one thing I liked about the other implementation was that you could bind to specific reducers on subscriptions.

So I decided to fork your wrapper and simply add a helper function that will allow for store subscription through a reducer filter, basically include this in your wrapper:
https://github.com/jprichardson/redux-watch

I also am using all types instances for my action payloads so I get TypeScript Intellisence which is nice, and all my Store typed instances extend my generic Store model:
https://github.com/born2net/studioEnterprise/blob/master/src/models/StoreModel.ts
that has all of Immutable.js goodness. So I am able to just extend it, and any instance that goes into the store becomes Immutably powered... if that's a word :)

Great job on finding the zone fix for the devtool

until the next, regards

Sean.

**Comment by born2net** [comment]
here is the code that has the built in filter per reducer / slice: https://github.com/born2net/angular2-redux-util/blob/master/src/app-store.ts

**Comment by rubyboy** [comment]
Beautiful. I love it!
Want to create a PR?

**Comment by born2net** [comment]
cool... I didn't connect my WebStorm to the fork as I ended up downloading the zip, so feel free to copy and paste if you like it... glad to contribute...

**Comment by rubyboy** [comment]
Cool, thanks for the tip.
I'll add the redux-watch bits to this library tomorrow. I think it's a great addition.

**Comment by born2net** [comment]
Question, do you think by putting the store inside the zone (so devtools will work) we will create more unnecessary event noise? even when using the onPush detection on components?

regards

Sean

**Comment by rubyboy** [comment]
Depends what do you mean by "noise" :)

It will most definitely not slow down anything or cause it to be inefficient, because the factory will only be called once if the provider is provided in the application bootstrapping. The way Angular 2 works is by using hierarchical injectors. If you put the provider at the top ("bootstrap") then it will be called once when the app starts. If you provided it lower down the chain (say, inside a component definition) then it may be called multiple times, which is not efficient but also not correct because that means multiple stores will be created (unless that's what you want...).

If the noise you're referring to is redux internal events, I don't think there are too many of them (there's the "init") and I think Angular should know about them, in case there is some affect on the state.

On the other hand, if the dev tools creates the noise (by pushing events through) then it is all done inside Angular's change detection, thus maybe affecting it. But, if that's happening then I guess you would want Angular to know about that anyway, to update UI (for example by cancelling actions in the dev tools). Anyway, this is only happening in development mode so I wouldn't worry about it.

So, bottom line, the answer is no. I don't think creating the app store inside Angular's umbrella is causing unnecessary noise nor affecting performance or functionality. Let me know if I'm wrong and we should rethink how to bootstrap the AppStore :)

**Comment by born2net** [comment]
I was referring to more change detection by ng2, but yes I agree with  your comments as they are inline with what I suspected... regards
